tests/e2e/sms-gateway.js

```diff
@@ -148,155 +148,65 @@ describe('sms-gateway api', () => {
           },
         ],
       };
-      pollSmsApi(body)
-        .then(done)
-        .catch(done.fail);
-      helper.handleUpdateModal();
```

Comment:
This old version looks really weird - there is clearly async stuff happening, but then the modal gets closed, which appears to cause the modal to happen immediately, likely before that `pollSmsApi` completes. I suppose the whole promise manager deal would make this work out, but seeing it here, I'm really glad we're able to move away from that!